modules/imaging_uploader/php/NDB_Menu_Filter_imaging_uploader.class.inc

```diff
@@ -674,13 +679,59 @@ class NDB_Menu_Filter_Imaging_Uploader extends NDB_Menu_Filter_Form
      */
     function toJSON()
     {
+        $maps = $this->getCandidateIDMapping();
+
+        $pscToCandID = $maps['pscids'];
+        $candToPSCID = $maps['candids'];
+
         $result            = $this->toArray();
         $result['form']    = $this->form->form;
         $result['mriList'] = $this->_getFileList($result['Data']);
+        $result['candToPSCID'] = $candToPSCID;
+        $result['pscToCandID'] = $pscToCandID;
 
         $result['maxUploadSize'] = Utility::getMaxUploadSize();
 
         return json_encode($result);
     }
+
+    /**
+     * Returns candidate PSCID / CandID arrays, or maps PSCID to CandID
+     * & vice versa if $reverse is true
+     *
+     * @param bool $reverse if true, return mappings
+     *
+     * @return array $result arrays or maps of candidate IDs
+     */
+    function getCandidateIDMapping($reverse = true)
```

Comment:
Generally, functions should only do one thing (see: Curly’s Law (https://blog.codinghorror.com/curlys-law-do-one-thing)).. the boolean argument adds complexity that, IMHO, should be handle by that function caller. It could return an array of the form : `[["PSCID"=>"AAA001", "CandID"=>"900001"],"PSCID"=>"AAA002", "CandID"=>"900002"],...]` and just array_walk/array_map to format it as required.
The other thing is that it will be called twice so there will be 2 identical databases calls. I would suggest to populate an instance variable with the pSelect results and the use array_map/array_walk on that variable.